### PR TITLE
Hoist common dependencies

### PR DESCRIPTION
**Pull request by drochgenius**
**WIP**

**ATTENTION: this update breaks all the unit tests, and I must find a solution before we can merge this**

Optimize dependency management so it's faster, more scalable as we will be adding more and more components.

For details on the approach read this: https://github.com/lerna/lerna/blob/master/doc/hoist.md

Execution time on my machine:

Execution timing on my Mac, with 6 packages in the repository.

| no hoist | hoist |
|:---------|:------|
| 692s     | 122s  |

You can calculate that it's 5.6 times faster.
Basically, without hoisting, installation time inscreases almost linearly with the number of packages.
With hoisting enabled, we install common dependencies once, and then just link them to all packages.

```shell
# hoisting enabled
npm run init  122.11s user 20.01s system 238% cpu 59.592 total
# no hoisting
npm run init  691.94s user 339.04s system 313% cpu 5:28.87 total
```

### CODE REVIEW

**Comment by drochgenius** [comment]
Hi @Dom-HMH-MTL , @adjamenhmh ,

I think we'll have to give up on this for the moment.
It doesn't play well at all with ES imports.
Works only with the Node module resolution algorithm,

In other words, the TS code builds fine for all packages, but demo screens and tests are borked.

**Comment by drochgenius** [comment]
I believe I have sufficiently tested this.
This is a major improvement to our development workflow.
I will now merge this to master.

**Comment by chrisunl** [comment]
I get this error on boot strapping:

> root@ bootstrap /Users/cmorris/Dev/assessement/mtl-assessment-components
> lerna bootstrap --hoist --nohoist=@hmh/component-base

lerna notice cli v3.4.3
lerna info Bootstrapping 7 packages
lerna info Installing external dependencies
lerna info hoist Installing hoisted dependencies into root
lerna info hoist Pruning hoisted dependencies
lerna info hoist Finished pruning hoisted dependencies
lerna ERR! npm install --global-style exited 239 in '@hmh/image-gallery'
lerna ERR! npm install --global-style stderr:
npm WARN @hmh/image-gallery@0.5.1 No repository field.

npm ERR! path /Users/cmorris/Dev/assessement/mtl-assessment-components/packages/image-gallery/node_modules/@hmh/component-base/node_modules/@polymer
npm ERR! code EEXIST
npm ERR! errno -17
npm ERR! syscall mkdir
npm ERR! EEXIST: file already exists, mkdir '/Users/cmorris/Dev/assessement/mtl-assessment-components/packages/image-gallery/node_modules/@hmh/component-base/node_modules/@polymer'
npm ERR! File exists: /Users/cmorris/Dev/assessement/mtl-assessment-components/packages/image-gallery/node_modules/@hmh/component-base/node_modules/@polymer
npm ERR! Move it away, and try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/cmorris/.npm/_logs/2018-10-31T14_58_29_474Z-debug.log

lerna ERR! npm install --global-style exited 239 in '@hmh/image-gallery'
lerna WARN complete Waiting for 3 child processes to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 239
npm ERR! root@ bootstrap: `lerna bootstrap --hoist --nohoist=@hmh/component-base`
npm ERR! Exit status 239
npm ERR!
npm ERR! Failed at the root@ bootstrap script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/cmorris/.npm/_logs/2018-10-31T14_58_29_593Z-debug.log
